fdbclient/include/fdbclient/RangeLock.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * RangeLock.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBCLIENT_RANGELOCK_H
+#define FDBCLIENT_RANGELOCK_H
+#pragma once
+
+#include "fdbclient/FDBTypes.h"
+#include "fdbrpc/fdbrpc.h"
+
+enum class RangeLockType : uint8_t {
+	Invalid = 0,
+	RejectAll = 1,
```

Comment:
Since only commit is blocked, maybe this should be called `RejectWrites` or `RejectCommits`?